### ISSUE TITLE
Change Request: Support Math

### ISSUE THREAD

**Issue by lumirlumir** [eslint/markdown#331]
### Environment

ESLint version: `v9.23.0`
@eslint/markdown version: `v6.3.0`
Node version: `v20.18.0`
npm version: `v10.9.2`
Operating System: Windows 11

### What problem do you want to solve?

Hello,

Math equation expressions using [LaTeX](https://en.wikipedia.org/wiki/LaTeX) are widely used in Markdown and are compatible with both CommonMark and GFM.  

Additionally, with [micromark-extension-math](https://github.com/micromark/micromark-extension-math?tab=readme-ov-file#micromark-extension-math) and [mdast-util-math](https://github.com/syntax-tree/mdast-util-math) available, integrating them into `@eslint/markdown` should be straightforward, just like we did in #328.  

Therefore, I’d like to propose adding support for Math syntax in `@eslint/markdown`.  

### What do you think is the correct solution?

Let's use [micromark-extension-math](https://github.com/micromark/micromark-extension-math?tab=readme-ov-file#micromark-extension-math) and [mdast-util-math](https://github.com/syntax-tree/mdast-util-math).

We can add an option to `languageOptions.math` as shown below, allowing it to be set to either `true` or `false`.  

```js
// eslint.config.js
import markdown from "@eslint/markdown";

export default [
    {
        files: ["**/*.md"],
        plugins: {
            markdown
        },
        language: "markdown/gfm",
        languageOptions: {
            frontmatter: true,
            math: true // It can be `true` or `false`.
        },
        rules: {
            "markdown/no-html": "error"
        }
    }
];
```

### Participation

- [x] I am willing to submit a pull request for this change.

**Comment by nzakas** [comment]
I'm not opposed, I just don't want to get into the habit of adding in every micromark extension just because it exists. In this case, I'm not sure that adding support for math actually adds to linting experience in any way. I doubt people will be writing rules to lint the mathematical formulas in the document.

We can leave this open to see if there's any further interest or use cases in the community. At this point, I'm not inclined to add this.

**Comment by nzakas** [comment]
Closing for now as I don't think there's a lot of demand for this. We can always revisit in the future if that changes.

**Comment by lumirlumir** [comment]
As `@eslint/markdown` grows and more error-prone syntax-detection rules are added, I've encountered some false positives while working on math equations in Markdown documents. So, I am kindly re-opening this issue.

While working on math equations (specifically, when I'm writing solutions for algorithm problem solving, e.g., on [Codeforces](https://codeforces.com/)), I've encountered the following kinds of false positives related to the `@eslint/markdown` rules.

For example:

### When writing permutation:

It leads to a false positive for the `markdown/no-missing-label-refs` rule.

- Code

    ```md
    $[1, 2, 3, 4]$
    ```

- Rendered

    $[1, 2, 3, 4]$

---

<img width="565" height="107" alt="Image" src="https://github.com/user-attachments/assets/0deb063e-9329-4d49-b34c-a25083d9424c" />

### When writing matrix-vector multiplication:

It leads to false positives for the `markdown/no-missing-label-refs` and `markdown/no-reversed-media-syntax` rules.

- Code

    ```md
    $(A \cdot x)[i] = \sum_{j=1}^{n} A[i][j] , x[j]$
    ```

- Rendered

    $(A \cdot x)[i] = \sum_{j=1}^{n} A[i][j] , x[j]$

---

<img width="741" height="101" alt="Image" src="https://github.com/user-attachments/assets/060e5b21-1209-4c74-88cd-87a2118db0f0" />

<img width="734" height="116" alt="Image" src="https://github.com/user-attachments/assets/9f892ecd-aca0-42c6-a43f-213cb77b2a19" />

### When writing convolution:

It leads to a false positive for the `markdown/no-space-in-emphasis` rule.

- Code

    ```md
    $(f * g)(x) + (h * k)(x)$
    ```

- Rendered

    $(f * g)(x) + (h * k)(x)$

---

<img width="577" height="109" alt="Image" src="https://github.com/user-attachments/assets/de94943e-8b1f-404f-9d0a-c39070ab98fa" />

---

For the reasons above, I would like to ask is it okay to add a `math` option to `@eslint/markdown`.

**Comment by fasttime** [comment]
I think adding support for math blocks (`$`&hellip;`$` and `$$\n`&hellip;`\n$$`) would be useful a useful addition, so I'm in favor of this proposal :+1:.  

To test compatibility, I included `mdast-util-math` and `micromark-extension-math` as parser extensions and mdast extensions as this issue suggests. All unit tests continued to pass without errors, indicating there should be little to no disruption to existing rules, and based on that, I don't expect this change to add any significant maintenance costs.
